langchain/src/vectorstores/singlestore.ts

```diff
@@ -4,18 +4,81 @@ import type {
   OkPacket,
   ResultSetHeader,
   FieldPacket,
+  PoolOptions,
 } from "mysql2/promise";
 import { format } from "mysql2";
+import { createPool } from "mysql2/promise";
+import { Metadata } from "@opensearch-project/opensearch/api/types.js";
 import { VectorStore } from "./base.js";
 import { Embeddings } from "../embeddings/base.js";
 import { Document } from "../document.js";
 
-export interface SingleStoreVectorStoreConfig {
-  connectionPool: Pool;
+export type DistanceMetrics = "DOT_PRODUCT" | "EUCLIDEAN_DISTANCE";
+
+const OrderingDirective: Record<DistanceMetrics, string> = {
+  DOT_PRODUCT: "DESC",
+  EUCLIDEAN_DISTANCE: "",
+};
+
+export interface ConnectionOptions extends PoolOptions {}
+
+type ConnectionWithUri = {
+  connectionOptions?: never;
+  connectionURI: string;
+};
+
+type ConnectionWithOptions = {
+  connectionURI?: never;
+  connectionOptions: ConnectionOptions;
+};
+
+type ConnectionConfig = ConnectionWithUri | ConnectionWithOptions;
+
+export type SingleStoreVectorStoreConfig = ConnectionConfig & {
   tableName?: string;
   contentColumnName?: string;
   vectorColumnName?: string;
   metadataColumnName?: string;
+  distanceMetrics?: DistanceMetrics;
```

Comment:
This should probably be singular since you're only picking one metric at a time, right? `distanceMetric`